File: day1.py
import math

# part 1
def fuel(parm):
    calc = math.floor(parm/3) - 2
    return calc

# part 2
def fuel_recursive(parm):
    calc = fuel(parm)
    if calc <= 0:
        return 0
    return calc + fuel_recursive(calc)

f = open("input1.txt", "r").readlines()

total = 0
for l in f:
    total += fuel_recursive(int(l))

print(total)

File: src/bin/day1.rs
use std::fs;

fn main() {
    let input: String = fs::read_to_string("input1.txt").expect("Couldn't open file");
    let total = input
        .lines()
        .map(|s| -> i32 { s.parse::<i32>().unwrap() })
        .fold(0, |acc, elt| acc + fuel_rec(elt));

    println!("{}", total);
}

fn fuel(mass: i32) -> i32 {
    (mass / 3) - 2
}

fn fuel_rec(mass: i32) -> i32 {
    let f: i32 = fuel(mass);

    if f < 0 {
        return 0;
    }

    f + fuel_rec(f)
}
